pkg/controller/v1beta1/inferenceservice/reconcilers/knative/ksvc_reconciler.go

```diff
@@ -224,25 +224,29 @@ func (r *KsvcReconciler) Reconcile() (*knservingv1.ServiceStatus, error) {
 		}
 		return nil, err
 	}
-	// Set ResourceVersion which is required for update operation.
-	desired.ResourceVersion = existing.ResourceVersion
-	// Add immutable annotations to avoid validation error during dry-run update.
-	desired.Annotations[knserving.CreatorAnnotation] = existing.Annotations[knserving.CreatorAnnotation]
-	desired.Annotations[knserving.UpdaterAnnotation] = existing.Annotations[knserving.UpdaterAnnotation]
-
-	// Do a dry-run update to avoid diffs generated because knative defaulter webhook is
-	// called when creating or updating the knative service. This will populate our local knative service object with any default values
-	// that are present on the remote version.
-	if err = r.client.Update(context.TODO(), desired, client.DryRunAll); err != nil {
-		log.Error(err, "Failed to perform dry-run create of knative service", "service", desired.Name)
-		return nil, err
-	}
 
 	err = retry.RetryOnConflict(retry.DefaultRetry, func() error {
 		log.Info("Updating knative service", "namespace", desired.Namespace, "name", desired.Name)
 		if err := r.client.Get(context.TODO(), types.NamespacedName{Name: desired.Name, Namespace: desired.Namespace}, existing); err != nil {
 			return err
 		}
+
+		// Set ResourceVersion which is required for update operation.
+		desired.ResourceVersion = existing.ResourceVersion
+		// Add immutable annotations to avoid validation error during dry-run update.
+		desired.Annotations[knserving.CreatorAnnotation] = existing.Annotations[knserving.CreatorAnnotation]
+		desired.Annotations[knserving.UpdaterAnnotation] = existing.Annotations[knserving.UpdaterAnnotation]
+
+		// Do a dry-run update to avoid diffs generated by default values introduced by knative's defaulter webhook.
+		// This will populate our local knative service object with any default values
+		// that are present on the remote version.
+		if err = r.client.Update(context.TODO(), desired, client.DryRunAll); err != nil {
+			// log only if it is not resource conflict error to avoid spamming
+			if !apierr.IsConflict(err) {
```

Comment:
Is the main change here to wrap setting the resource version in the retry block ?

Reply:
Also do we still need the outer get ksvc call if it is wrapped in the retry block